internal/cmd/docker/push.go

```diff
@@ -0,0 +1,110 @@
+package docker
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/registry"
+	"github.com/docker/docker/client"
+	cmds "github.com/saucelabs/saucectl/internal/cmd"
+	"github.com/saucelabs/saucectl/internal/segment"
+	"github.com/saucelabs/saucectl/internal/usage"
+	"github.com/spf13/cobra"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+func PushCommand() *cobra.Command {
+	var timeout time.Duration
+	var quiet bool
+
+	cmd := &cobra.Command{
+		Use:          "push <repo> <image_name>",
+		Short:        "Push a Docker image to the Sauce Labs Container Registry.",
+		SilenceUsage: true,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 || args[0] == "" {
+				return errors.New("no repo name specified")
+			}
+			if len(args) == 1 || args[1] == "" {
+				return errors.New("no docker image specified")
+			}
+
+			return nil
+		},
+		PreRun: func(cmd *cobra.Command, args []string) {
+			tracker := segment.DefaultTracker
+
+			go func() {
+				tracker.Collect(
+					cases.Title(language.English).String(cmds.FullName(cmd)),
+					usage.Properties{}.SetFlags(cmd.Flags()),
+				)
+				_ = tracker.Close()
+			}()
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			repo := args[0]
+			image := args[1]
+			auth, err := imageRunnerService.RegistryLogin(context.Background(), repo)
+			if err != nil {
+				return fmt.Errorf("failed to fetch auth token: %v", err)
+			}
+			return pushDockerImage(image, auth.Username, auth.Password, timeout, quiet)
+		},
+	}
+
+	flags := cmd.PersistentFlags()
+	flags.DurationVar(&timeout, "timeout", 5*time.Minute, "Configure the timeout duration for docker push.")
```

Comment:
Default value seems not to be necessary in the description. It's generated automatically. 

```
      --timeout duration   Configure the timeout duration for docker push. Default: 5 minutes. (default 5m0s)
```